projects/batfish/src/test/java/org/batfish/job/BatfishJobExecutorTest.java

```diff
@@ -0,0 +1,218 @@
+package org.batfish.job;
+
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import org.batfish.common.BatfishException;
+import org.batfish.common.BatfishLogger;
+import org.batfish.common.BatfishLogger.BatfishLoggerHistory;
+import org.batfish.common.CompositeBatfishException;
+import org.batfish.common.util.CommonUtil;
+import org.batfish.config.Settings;
+import org.batfish.datamodel.answers.AnswerElement;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/** Tests for {@link BatfishJobExecutor}. */
+public class BatfishJobExecutorTest {
+
+  @Rule public ExpectedException _thrown = ExpectedException.none();
+  BatfishLogger _logger;
+
+  @Before
+  public void setup() {
+    _logger = new BatfishLogger("info", false);
+  }
+
+  @Test
+  public void testExecuteJobsResults() {
+    Settings settings = new Settings();
+    settings.setHaltOnParseError(false);
+    settings.setSequential(false);
+
+    List<BfTestJob> jobs = new ArrayList<>();
+    jobs.add(new BfTestJob(settings, "result1"));
+    jobs.add(new BfTestJob(settings, "result2"));
+
+    Set<String> output = new HashSet<>();
+    BfTestAnswerElement ae = new BfTestAnswerElement();
+    BatfishJobExecutor<BfTestJob, BfTestAnswerElement, BfTestResult, Set<String>> executor =
+        new BatfishJobExecutor<>(
+            settings, _logger, settings.getHaltOnParseError(), "Test Job Executor");
+    executor.executeJobs(jobs, output, ae);
+
+    // checking the outputs produced by the tasks
+    assertEquals(output, Sets.newHashSet("result1", "result2"));
+  }
+
+  @Test
+  public void testExecuteJobsTime() {
+    Settings settings = new Settings();
+    settings.setHaltOnParseError(false);
+
+    List<BfTestJob> jobs = new ArrayList<>();
+    jobs.add(new BfTestJob(settings, "result1"));
+    jobs.add(new BfTestJob(settings, "result2"));
+
+    Set<String> output = new HashSet<>();
+    BfTestAnswerElement ae = new BfTestAnswerElement();
+    BatfishJobExecutor<BfTestJob, BfTestAnswerElement, BfTestResult, Set<String>> executor =
+        new BatfishJobExecutor<>(
+            settings, _logger, settings.getHaltOnParseError(), "Test Job Executor");
+    long startTime = System.currentTimeMillis();
+    executor.executeJobs(jobs, output, ae);
+    long elapsedTime = System.currentTimeMillis() - startTime;
+
+    // tasks executed in parallel should take less than sequential time in most normal cases
+    assertThat(elapsedTime, lessThanOrEqualTo(2000L));
```

Comment:
It is not clear to me that sequential time would take more than 2000 milliseconds. The improvement you have implemented is over polling, not over sequential execution. Polling has been totally eliminated, so the test assertion is of dubious value here IMO.

Reply:
I should add that this test *is* actually useful to compare to polling. But then you should adjust the comment accordingly.

Reply:
The BfTestJob is not using any parsing code, it is just returning a string after a configurable delay. So the tests are independent of any parsing logic and they are only testing the  BfJObExecutor

Reply:
Ah I misunderstood the test. I withdraw my objection.